File: Python/konversi_suhu.py
print("\t" + "======= | Konversi Suhu | =======")
print("\t\t" + "1. Celcius")
print("\t\t" + "2. Fahrenheit")
print("\t\t" + "3. Reamur")
print("\t\t" + "4. Kelvin")
print("\t" + "=================================")

while True:
    jenis_awal = int(input("\n\t" + "Pilih Jenis Suhu (1, 2, 3, 4) = "))
    
    if jenis_awal == 1:
        jumlah_suhu = float(input("\n\t" + "Masukkan Jumlah Suhu = "))
        jenis_akhir = int(input("\n\t" + "Konversi Ke (1, 2, 3, 4) = "))
        if jenis_akhir == 2:
            hasil_konversi = jumlah_suhu * (9/5) + 32
            print("\n\t" + str(jumlah_suhu) + "°C" + " = " + str(hasil_konversi) + "°F")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        elif jenis_akhir == 3:
            hasil_konversi = jumlah_suhu * (4/5)
            print("\n\t" + str(jumlah_suhu) + "°C" + " = " + str(hasil_konversi) + "°R")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        elif jenis_akhir == 4:
            hasil_konversi = jumlah_suhu + 273.15
            print("\n\t" + str(jumlah_suhu) + "°C" + " = " + str(hasil_konversi) + "K")
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        else:
            print("\n\t" + "Invalid Input!!!")
    elif jenis_awal == 2:
        jumlah_suhu = float(input("\n\t" + "Masukkan Jumlah Suhu = "))
        jenis_akhir = int(input("\n\t" + "Konversi Ke (1, 2, 3, 4) = "))
        if jenis_akhir == 1:
            hasil_konversi = (jumlah_suhu - 32) * (5/9)
            print("\n\t" + str(jumlah_suhu) + "°F" + " = " + str(hasil_konversi) + "°C")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        elif jenis_akhir == 3:
            hasil_konversi = (jumlah_suhu - 32) * (4/9)
            print("\n\t" + str(jumlah_suhu) + "°F" + " = " + str(hasil_konversi) + "°R")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        elif jenis_akhir == 4:
            hasil_konversi = (jumlah_suhu - 32) * (5/9) + 273.15
            print("\n\t" + str(jumlah_suhu) + "°F" + " = " + str(hasil_konversi) + "K")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        else:
            print("\n\t" + "Invalid Input!!!")
    elif jenis_awal == 3:
        jumlah_suhu = float(input("\n\t" + "Masukkan Jumlah Suhu = "))
        jenis_akhir = int(input("\n\t" + "Konversi Ke (1, 2, 3, 4) = "))
        if jenis_akhir == 1:
            hasil_konversi = jumlah_suhu * (5/4)
            print("\n\t" + str(jumlah_suhu) + "°R" + " = " + str(hasil_konversi) + "°C")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        elif jenis_akhir == 2:
            hasil_konversi = jumlah_suhu * (9/4) + 32
            print("\n\t" + str(jumlah_suhu) + "°R" + " = " + str(hasil_konversi) + "°F")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        elif jenis_akhir == 4:
            hasil_konversi = jumlah_suhu * (5/4) + 273.15
            print("\n\t" + str(jumlah_suhu) + "°R" + " = " + str(hasil_konversi) + "K")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        else:
            print("\n\t" + "Invalid Input!!!")
    elif jenis_awal == 4:
        jumlah_suhu = float(input("\n\t" + "Masukkan Jumlah Suhu = "))
        jenis_akhir = int(input("\n\t" + "Konversi Ke (1, 2, 3, 4) = "))
        if jenis_akhir == 1:
            hasil_konversi = jumlah_suhu - 273.15
            print("\n\t" + str(jumlah_suhu) + "K" + " = " + str(hasil_konversi) + "°C")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        elif jenis_akhir == 2:
            hasil_konversi = (jumlah_suhu - 273.15) * (9/5) + 32
            print("\n\t" + str(jumlah_suhu) + "K" + " = " + str(hasil_konversi) + "°F")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        elif jenis_akhir == 3:
            hasil_konversi = (jumlah_suhu - 273.15) * (4/5)
            print("\n\t" + str(jumlah_suhu) + "K" + " = " + str(hasil_konversi) + "°R")
            
            ulangi = input("\n\tIngin Mengulangi Kembali? (y/n) = ")
            if ulangi == "y":
                pass
            elif ulangi == "n":
                break
            else:
                print("\n\tInvalid Input!!!")
                break
        else:
            print("\n\t" + "Invalid Input!!!")
            break
    else:
        print("\n\t" + "Invalid Input!!!")
        break

File: C++/Konversi Suhu.cpp
#include <iostream>
#include <cstdlib>
#include <string>

#ifdef _WIN32
#include <windows.h>
#else
#include <sys/ioctl.h>
#include <unistd.h>
#endif

int getTerminalWidth() {
    int width = 50;
    
    #ifdef _WIN32
    CONSOLE_SCREEN_BUFFER_INFO csbi;
    GetConsoleScreenBufferInfo(GetStdHandle(STD_OUTPUT_HANDLE), &csbi);
    width = csbi.srWindow.Right - csbi.srWindow.Left + 1;
    #else
    struct winsize w;
    ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
    width = w.ws_col;
    #endif
    
    return width;
}

using namespace std;

int main() {
    system("clear");
    
    int screenWidth = getTerminalWidth();
    
    while (true) {
        cout << string(screenWidth, '-') << endl << endl;
        cout << string((screenWidth - 17) / 2, ' ') << "\033[1;36m# KONVERSI SUHU #\033[0m" << endl << endl;
        cout << string((screenWidth - 31) / 2, ' ') << "-------------------------------" << endl << endl;
        cout << string((screenWidth - 31) / 2, ' ') << "\033[1;32m(1)\033[0m. Celcius    \033[1;32m(2)\033[0m. Fahrenheit" << endl << endl;
        cout << string((screenWidth - 31) / 2, ' ') << "\033[1;32m(3)\033[0m. Réamur     \033[1;32m(4)\033[0m. Kelvin" << endl << endl;
        cout << string((screenWidth - 11) / 2, ' ') << "\033[1;31m(5)\033[0m. Keluar" << endl << endl;
        cout << string(screenWidth, '-') << endl << endl;
        
        int initialTemperatureType, finalTemperatureType;
        double formula;
        string temperatureValue;
        
        cout << "Silahkan pilih jenis suhu awal (1, 2, 3, 4) : ";
        cin >> initialTemperatureType;
        
        if (initialTemperatureType == 1) {
            cout << endl << "Silahkan masukkan jumlah suhu : ";
            cin >> temperatureValue;
            
            bool isNumber = true;
            for (int i = 0; i < temperatureValue.length(); i++) {
                if (!isdigit(temperatureValue[i])) {
                    isNumber = false;
                }
            }
            
            if (isNumber == true) {
                cout << endl << "Silahkan pilih jenis suhu akhir (1, 2, 3, 4) : ";
                cin >> finalTemperatureType;
                
                if (finalTemperatureType == 1) {
                    cout << endl << "\033[1;33mJenis suhu tidak boleh sama!\033[0m" << endl;
                } else if (finalTemperatureType == 2) {
                    int integerTemperatureValue = stoi(temperatureValue);
                    formula = (integerTemperatureValue * 9 / 5 + 32);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "°C" << " = " << formula << "°F" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 3) {
                    int integerTemperatureValue = stoi(temperatureValue);
                    formula = (integerTemperatureValue * 4 / 5);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "°C" << " = " << formula << "°Ré" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 4) {
                    int integerTemperatureValue = stoi(temperatureValue);
                    formula = (integerTemperatureValue + 273.15);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "°C" << " = " << formula << "K" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 5) {
                    cout << endl;
                    break;
                } else {
                    cout << endl << "\033[1;31mInvalid Type!\033[0m" << endl << endl;
                    break;
                }
            } else {
                cout << endl << "\033[1;31mInvalid Value!\033[0m" << endl << endl;
                break;
            }
        } else if (initialTemperatureType == 2) {
            cout << endl << "Silahkan masukkan jumlah suhu : ";
            cin >> temperatureValue;
            
            bool isNumber = true;
            for (int i = 0; i < temperatureValue.length(); i++) {
                if (!isdigit(temperatureValue[i])) {
                    isNumber = false;
                }
            }
            
            if (isNumber == true) {
                cout << endl << "Silahkan pilih jenis suhu akhir (1, 2, 3, 4) : ";
                cin >> finalTemperatureType;
                
                if (finalTemperatureType == 1) {
                    int integerTemperatureValue = stoi(temperatureValue);
                    formula = ((integerTemperatureValue - 32) * 5 / 9);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "°F" << " = " << formula << "°C" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 2) {
                    cout << endl << "\033[1;33mJenis suhu tidak boleh sama!\033[0m" << endl;
                } else if (finalTemperatureType == 3) {
                    int integerTemperatureValue = stoi(temperatureValue);
                    formula = ((integerTemperatureValue - 32) * 4 / 9);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "°F" << " = " << formula << "°Ré" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 4) {
                    int integerTemperatureValue = stoi(temperatureValue);
                    formula = ((integerTemperatureValue + 459.67) * 5 / 9);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "°F" << " = " << formula << "K" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 5) {
                    cout << endl;
                    break;
                } else {
                    cout << endl << "\033[1;31mInvalid Type!\033[0m" << endl << endl;
                    break;
                }
            } else {
                cout << endl << "\033[1;31mInvalid Value!\033[0m" << endl << endl;
                break;
            }
        } else if (initialTemperatureType == 3) {
            cout << endl << "Silahkan masukkan jumlah suhu : ";
            cin >> temperatureValue;
            
            bool isNumber = true;
            for (int i = 0; i < temperatureValue.length(); i++) {
                if (!isdigit(temperatureValue[i])) {
                    isNumber = false;
                }
            }
            
            if (isNumber == true) {
                cout << endl << "Silahkan pilih jenis suhu akhir (1, 2, 3, 4) : ";
                cin >> finalTemperatureType;
                
                if (finalTemperatureType == 1) {
                    int integerTemperatureValue = stoi(temperatureValue);
                    formula = (integerTemperatureValue * 5 / 4);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "°Ré" << " = " << formula << "°C" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 2) {
                    int integerTemperatureValue = stoi(temperatureValue);
                    formula = (integerTemperatureValue * 9 / 4 + 32);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "°Ré" << " = " << formula << "°F" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 3) {
                    cout << endl << "\033[1;33mJenis suhu tidak boleh sama!\033[0m" << endl;
                } else if (finalTemperatureType == 4) {
                    int integerTemperatureValue = stoi(temperatureValue);
                    formula = (integerTemperatureValue * 5 / 4 + 273.15);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "°Ré" << " = " << formula << "K" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 5) {
                    cout << endl;
                    break;
                } else {
                    cout << endl << "\033[1;31mInvalid Type!\033[0m" << endl << endl;
                    break;
                }
            } else {
                cout << endl << "\033[1;31mInvalid Value!\033[0m" << endl << endl;
                break;
            }
        } else if (initialTemperatureType == 4) {
            cout << endl << "Silahkan masukkan jumlah suhu : ";
            cin >> temperatureValue;
            
            if (isdigit(temperatureValue[0])) {
                cout << endl << "Silahkan pilih jenis suhu akhir (1, 2, 3, 4) : ";
                cin >> finalTemperatureType;
                
                if (finalTemperatureType == 1) {
                    double doubleTemperatureValue = stod(temperatureValue);
                    formula = (doubleTemperatureValue - 273.15);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "K" << " = " << formula << "°C" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 2) {
                    double doubleTemperatureValue = stod(temperatureValue);
                    formula = (doubleTemperatureValue * 9 / 5 - 459.67);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "K" << " = " << formula << "°F" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 3) {
                    double doubleTemperatureValue = stod(temperatureValue);
                    formula = ((doubleTemperatureValue - 273.15) * 4 / 5);
                    system("clear");
                    cout << string(screenWidth, '-') << endl << endl;
                    cout << string((screenWidth - 12) / 2, ' ') << "\033[1;32m" << temperatureValue << "K" << " = " << formula << "°Ré" << "\033[0m" << endl << endl;
                    cout << string(screenWidth, '-') << endl;
                } else if (finalTemperatureType == 4) {
                    cout << endl << "\033[1;33mJenis suhu tidak boleh sama!\033[0m" << endl;
                } else if (finalTemperatureType == 5) {
                    cout << endl;
                    break;
                } else {
                    cout << endl << "\033[1;31mInvalid Type!\033[0m" << endl << endl;
                    break;
                }
            } else {
                cout << endl << "\033[1;31mInvalid Value!\033[0m" << endl << endl;
                break;
            }
        } else {
            cout << endl << "\033[1;31mInvalid Type!\033[0m" << endl << endl;
            break;
        }
        
        char repeat;
        cout << endl << "Ingin melakukan konversi lagi? (\033[1;32my\033[0m/\033[1;31mn\033[0m) : ";
        cin >> repeat;
        
        if (tolower(repeat) == 'y') {
            system("clear");
        } else if (tolower(repeat) == 'n') {
            cout << endl;
            break;
        } else {
            cout << endl << "\033[1;31mInvalid Input!\033[0m" << endl << endl;
            break;
        }
    }
    return 0;
}